llama_stack/providers/remote/inference/ollama/ollama.py

```diff
@@ -193,6 +191,41 @@ async def health(self) -> HealthResponse:
         except Exception as e:
             return HealthResponse(status=HealthStatus.ERROR, message=f"Health check failed: {str(e)}")
 
+    async def check_model_availability(self, model: str) -> bool:
+        """
+        Check if a specific model is available in Ollama.
+
+        :param model: The model identifier to check.
+        :return: True if the model is available, False otherwise.
+        """
+        try:
+            available_models = await self._query_available_models()
+            return model in available_models
+        except Exception as e:
+            logger.error(f"Error checking model availability: {e}")
+            return False
+
+    async def _query_available_models(self) -> list[str]:
```

Comment:
how does it interact with the `list_models()` method implemented above? given that the registry helper is now added as a mixin (and has its own list_models() method), it makes things a bit confusing. do you need to add the registry helper as a mixin (for now)?

Reply:
sorry I lost track of this PR. yeah second option seems reasonable.